tests/units/partialsync/test_partial_sync_utils.py

```diff
@@ -0,0 +1,112 @@
+from unittest import TestCase, mock
+from tempfile import TemporaryDirectory
+
+from pipelinewise.fastsync.partialsync.utils import load_into_snowflake, upload_to_s3, update_state_file
+from tests.units.partialsync.utils import PartialSync2SFArgs
+
+
+class PartialSyncUtilsTestCase(TestCase):
+    """Test case for partial sync utils"""
+
+    def test_upload_to_s3(self):
+        """Test _upload_to_s3 method"""
+        with TemporaryDirectory() as temp_test_dir:
+            test_file_part = f'{temp_test_dir}/foo.gz1'
+            test_s3_key = 'foo_s3_key'
+            with open(test_file_part, 'w', encoding='utf8') as file_to_test:
+                file_to_test.write('bar')
+
+            class MockedSnowflake(PartialSyncUtilsTestCase):
+                """mock for snowflake"""
+                def upload_to_s3(self, file_part, tmp_dir):
+                    """mock for upload_to_s3 method"""
+                    self.assertEqual(tmp_dir, temp_test_dir)
+                    self.assertEqual(file_part, test_file_part)
+                    return test_s3_key
```

Comment:
I don't understand this code, this class is both a mock and a test class at the same time !?
To mock classes, please use `Mock` and keep asserts inside under the unit test method `test_upload_to_s3`

Reply:
- [ ] better way for mocking